src/lib/render-value.ts

```diff
@@ -1,12 +1,23 @@
 export const renderValue = (value: unknown): string[] => {
 	if (Array.isArray(value)) {
-		return ["Array", `[${value.length} elements]`];
+		return [
+			"Array",
+			value.length
+				? value.length === 1
+					? `[${value.length} element]`
+					: `[${value.length} elements]`
+				: "[]",
+		];
+	}
+
+	if (value instanceof Object && Object.keys(value).length === 0) {
+		return ["Object", "{}"];
 	}
 
 	if (typeof value === "object" && value !== null) {
 		const keys = Object.keys(value);
 		return [
-			(value as { type: string })?.type,
+			(value as { type: string })?.type ?? "Object",
```

Comment:
```suggestion
			(value as { type?: string })?.type ?? "Object",
```
in case we guess type is optional we might want to specify during assertion as well?

Reply:
Marking the type as optional was not part of this PR. I’ve simply added the nullish coalescing operator and set it to return "object" when the type is missing. Since we're already asserting the value, this change won’t have any impact. However, if needed, we can still mark the type as optional since we are doing optional chaining.